contracts/PriceOracle.sol

```diff
@@ -100,6 +104,7 @@ abstract contract PriceOracle {
 
         // Calculate elapsed time
         uint256 elapsedTime = block.timestamp - snapshot.lastUpdated;
+        // Require at least one block since last update
```

Comment:
This is key and proper. It implies that we cannot have this transaction happen when the block previously contained a transaction that involves an update to the oracle. Holds for both types of oracles. We need to be aware of this.

Reply:
We compare based on time and it happens earlier.